R/standardise_titles.R

```diff
@@ -61,17 +61,52 @@ standardise_titles <- standardize_titles <- function(s, strict = FALSE, api_key
   }
   out <- out$out
   }
+  
+  # Step three: standardise strings returned
   # Transforms strings to ASCII character encoding
   out <- suppressWarnings(stringi::stri_trans_general(out, id = "Latin-ASCII"))
+  # standardises NAs
   out[out == "NANA"] <- NA
-  out <- trimws(out)
   out <- gsub("\\.(?=\\.*$)", "", out, perl = TRUE)
+  # standardises spaces before and after apostrophes and comma spaces
+  out <- gsub(" '|' ","'", out)
+  # Delete hyphens when separating two parts of the title
+  # (when there is a space before and after)
+  out <- gsub(" - ", " ", out)
+  # Delete special character found in some treaty titles
+  out <- gsub("\U00AC[[:alpha:]]{1}\\s|\U00AC\\s", "", out)
+  # Add space after a comma
+  out <- textclean::add_comma_space(out)
+  # standardises some country abbreviations
   out <- gsub("U.K.", "UK", out)
   out <- gsub("U.S.S.R.", "USSR", out)
-  out <- gsub("U.S. ", "USA", out)
+  out <- gsub("U.S. ", "USA ", out)
+  # standardises some specific word spellings
   out <- gsub("Art\\.", "Article", out)
   out <- gsub("\\#", "Number ", out)
-  out <- textclean::add_comma_space(out)
+  out <- gsub("co-operation|coperation", "Cooperation", out, ignore.case = TRUE)
+  out <- gsub("co-operative|coperative", "Cooperative", out, ignore.case = TRUE)
+  out <- gsub("wild life|wild-life", "Wildlife", out, ignore.case = TRUE)
+  out <- gsub("Decision Making", "Decision-Making", out, ignore.case = TRUE)
+  out <- gsub("MaasMeuse", "Maas", out, ignore.case = TRUE)
+  out <- ifelse(stringr::str_detect(out, "Test-Ban|Foot-and-Mouth|Nuclear-Weapon-Free|Public-Participation|Deep-Sea"),
+                gsub("-", " ", out), out)
+  out <- gsub("land-based|landbased", "Land Based", out, ignore.case = TRUE)
+  out <- gsub("Vietnam", "Viet Nam", out, ignore.case = TRUE)
+  out <- gsub("shipsballast|ship's ballast", "ships ballast", out, ignore.case = TRUE)
+  # standardises regions spelling
+  out <- gsub("North-East|Northeast", "North East", out, ignore.case = TRUE)
+  out <- gsub("North-Eastern|Northeastern", "North Eastern", out, ignore.case = TRUE)
+  out <- gsub("North-West|Northwest", "North West", out, ignore.case = TRUE)
+  out <- gsub("North-western|Northwestern", "North Western", out, ignore.case = TRUE)
+  out <- gsub("South-East|Southeast", "South East", out, ignore.case = TRUE)
+  out <- gsub("South-Eastern|Southeastern", "South Eastern", out, ignore.case = TRUE)
+  out <- gsub("South-West|Southwest", "South West", out, ignore.case = TRUE)
+  out <- gsub("South-Western|Southwestern", "South Western", out, ignore.case = TRUE)
+  out <- gsub("Indo-Pacific|Indopacific|Asia-Pacific|Asiapacific", "Asia Pacific", out, ignore.case = TRUE)
```

Comment:
Can we break some of this down into helper functions (for next release) to help with debugging, code readability, etc?

Reply:
Yes it can be done for next release. I just created an issue on this.